### PR TITLE
fix(applications/web): prevent duplicate unique roles in a project team (BOAS-1553)

### PR DESCRIPTION
**Pull request by BenSurgisonPINS**
## Describe your changes

- Make sure unique roles are not available in the choose/change role page when they have already been filled within the team.
- Update unit tests to handle and prove support of unique roles
- Manually tested the functionality

## BOAS-1553 Unique roles within a project team such as "Case manager" can be assigned to more than one member of the project team.
https://pins-ds.atlassian.net/browse/BOAS-1553

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/server/applications/case/project-team/applications-project-team.controller.js [hunk 1]
```diff
@@ -18,18 +18,43 @@ import {
...
+ *
+ * Available roles include:
+ * - The role already assigned to the team member
+ * - The roles that can be assigned to multiple team members
+ * - The unassigned roles that cannot be assigned to multiple team members
+ *
+ * @param {Partial<ProjectTeamMember>} projectTeamMember
+ * @param {Partial<ProjectTeamMember>[]} projectTeamMembers
+ */
+function availableRoles(projectTeamMember, projectTeamMembers) {
```
Comment (francescobrenci-AR) [inline apps/web/src/server/applications/case/project-team/applications-project-team.controller.js:44]: this function and the `const` on the next line have the same name - i'd use different ones

**Review by francescobrenci-AR — COMMENTED** [review]
LGTM just a small comment

File: apps/web/src/server/applications/case/project-team/applications-project-team.controller.js (hunk 1 above)
Comment (francescobrenci-AR) [inline apps/web/src/server/applications/case/project-team/applications-project-team.controller.js:44]: this function and the const on the next line have the same name - i'd use different onew

Comment (BenSurgisonPINS) [reply apps/web/src/server/applications/case/project-team/applications-project-team.controller.js:44]: Good point, I've fixed as requested.

**Pull request merged**